src/Microsoft.Diagnostics.Monitoring.EventPipe/Logs/TraceEventExtensions.cs

```diff
@@ -26,21 +26,29 @@ public static void GetLogMessageJsonEventData(
             string argsJson = (string)traceEvent.PayloadValue(6);
             string formattedMessage = (string)traceEvent.PayloadValue(7);
 
-            string? activityTraceId;
-            string? activitySpanId;
-            string? activityTraceFlags;
-            if (traceEvent.Version >= 2)
-            {
-                // Note: Trace correlation fields added in .NET 9
-                activityTraceId = (string)traceEvent.PayloadValue(8);
-                activitySpanId = (string)traceEvent.PayloadValue(9);
-                activityTraceFlags = (string)traceEvent.PayloadValue(10);
-            }
-            else
+            string? activityTraceId = null;
+            string? activitySpanId = null;
+            string? activityTraceFlags = null;
+
+            // NOTE: The Microsoft-Extensions-Logging EventSource is created with
+            // EventSourceSettings.EtwSelfDescribingEventFormat (TraceLogging). In TraceLogging,
+            // the ETW event header Version is always 0 even if the [Event] attribute declares
+            // Version=2. The three correlation fields (ActivityTraceId, ActivitySpanId,
+            // ActivityTraceFlags) were added in .NET 9, and with TraceLogging their presence
+            // is discoverable only via payload schema (field names), not header Version.
+            // Therefore we detect by payload name instead of checking traceEvent.Version.
+            int idxTraceId = traceEvent.PayloadIndex("ActivityTraceId");
```

Comment:
Previously, it looked like it didn't matter whether activityTraceId was present in order to try to extract SpanId and TraceFlags. Now it looks to be explicit that if ActivityTraceId is absent, that it doesn't try to extract SpanId or TraceFlags. I imagine that the expectation is that all 3 will be present or absent in tandem, but would PayloadByName work?
```C#
activityTraceId = (string)traceEvent.PayloadByName("ActivityTraceId");
activitySpanId = (string)traceEvent.PayloadByName("ActivitySpanId");
activityTraceFlags = (string)traceEvent.PayloadByName("ActivityTraceFlags");
```
https://github.com/microsoft/perfview/blob/c63bac3ef8827fed6b6a4cdb6cdf32234daa6d8d/src/TraceEvent/TraceEvent.cs#L1270-L1279

Reply:
Good point - I'm updating it to call `PayloadByName` for each field individually to restore the existing behavior of extracting what's available.